app/src/main/java/com/example/myapplicationwork/util/ResProvider.kt

```diff
@@ -1,12 +1,11 @@
 package com.example.myapplicationwork.util
 
-import android.content.Context
+import com.example.myapplicationwork.GetInfoFromApi
 import com.example.myapplicationwork.data.Movie
-import com.example.myapplicationwork.data.loadMovies
 
-class ResProvider(val context: Context): Provider {
+class ResProvider() : Provider {
     override suspend fun loadFilms(): List<Movie> {
-        return loadMovies(context)
+        val getInfoFromApi = GetInfoFromApi()
```

Comment:
let's `getInfoFromApi` variable as global
and create it via lazy or inside init block